### PR TITLE
Move RCs creation from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by marcussc**
## Description
Adds a Workflow that will be triggered when a release tag is created, which builds the app and uploads to TestFlight and Mobile Device Cloud.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12568

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -274,37 +274,6 @@ jobs:
 
 workflows:
   version: 2
-  beta-release:
-    jobs:
-      - dependencies:
-          filters:
-            tags:
-              only:
```
Comment (pascal-brause) [inline .circleci/config.yml:282]: is this filter missing in the new config?

File: .github/workflows/ReleaseCandidate.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+name: Create Release Candidate
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+jobs:
+  dependencies:
+    runs-on: self-hosted
```
Comment (pascal-brause) [inline .github/workflows/ReleaseCandidate.yml:9]: ❤️

File: src/xcode/fastlane/Matchfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-git_url("git@github.com:corona-warn-app/cwa-certificates.git")
+git_url("https://github.com/corona-warn-app/cwa-certificates.git")
```
Comment (pascal-brause) [inline src/xcode/fastlane/Matchfile:1]: ssh does not work anymore?

Comment (marcussc) [reply src/xcode/fastlane/Matchfile:1]: Nope ssh seems blocked on the firewall

File: .circleci/config.yml (hunk 1 above)
Comment (marcussc) [reply .circleci/config.yml:282]: https://github.com/corona-warn-app/cwa-app-ios/pull/4429/files/8ff2d28fe57d2546c3050e6163b0198b743d5748#diff-150d88cc502b284bc9405bca3a8d7aa5c2f45f6a44dddeb994ec9843bc4236e0R5 nope :D

Comment (pascal-brause) [reply .circleci/config.yml:282]: ah okay I did not see this

File: src/xcode/fastlane/Matchfile (hunk 3 above)
Comment (pascal-brause) [reply src/xcode/fastlane/Matchfile:1]: oki

**Review by pascal-brause — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**